public/javascripts/index.js

```diff
@@ -0,0 +1,269 @@
+
+    $.validator.addMethod("pollinterval", function (value, element) {
+        const v = $("input[name=\"" + element.name + "\"]").val();
+        return (v == 0 || (v >= 60 && v <= 600));
+    }, "Enter interval in seconds 0 or min. 60 - max 600");
+
+    $("#xcoobee_demo_form")
+        .submit(e => e.preventDefault())
+        .validate({
+            submitHandler: function () {
+                alertBlock.hide();
+                const submitButtonName = $(this.submitButton).attr("name");
+                const pollInterval = +$("#xcoobee_poll_interval").val() || 0;
+                if (submitButtonName === "start_poll" && pollInterval !== 0) {
+                    $("#start_poll").removeClass("btn-outline-primary").addClass("btn-primary").prop("disabled", true);
+                    $("#stop_poll").removeClass("btn-light").addClass("btn-outline-primary");
+                    $("#manual_poll").prop("disabled", true);
+                    displayPollerResponse();
+                    pollServer(pollInterval);
+                }
+                if (submitButtonName === "manual_poll" || pollInterval === 0) {
+                    $("button").prop("disabled", true);
+                    displayPollerResponse().then(() => stopPoller());
+                }
+                return false;
+            },
+        });
+
+    $("#stop_poll").on("click", () => {
+        stopPoller();
+        alertBlock.hide();
+    });
+
+    // ============= Payments page ===========
+    $("#payment_form").submit(e => e.preventDefault());
+    $("#request_payment").click(function () {
+        const properties = {};
+        const formData = $(this.form).serializeArray();
+        formData.forEach((item) => {
+            properties[item.name] = item.value;
+        });
+
+        const xcoobeeSDK = getPaymentSDKInstance(properties.external_device_id);
+
+        const link = xcoobeeSDK.createPayUrl({
+            amount: parseFloat(properties.payment_amount),
+            reference: properties.payment_reference,
+        });
+
+        $("#payment_cycle").attr("src", link);
+        setTimeout(() => pollServer(2), 25000);
+    });
+
+    // ============= Invoice page ===========
+    $("#invoice_form")
+        .submit(e => e.preventDefault())
+        .validate({
+            submitHandler: function () {
+                const settings = getCookie(COOKIE_NAME) || {};
+
+                if (!settings.campaign_id) {
+                    alertBlock.show();
+                    handleError("Campaign ID is required");
+                    return;
+                }
+
+                const properties = {};
+                const formData = $("#invoice_form").serializeArray();
+                formData.forEach((item) => {
+                    properties[item.name] = item.value;
+                });
+
+                window.open(`/generate-invoice?ref=${properties.payment_reference}&amount=${properties.payment_amount}&campaignId=${settings.campaign_id}`, "_blank");
```

Comment:
To be on the safe side, I would use `encodeUriComponet` for all these parameters even if we know they are safe ahead of time.